applications/API/models.py

```
@@ -0,0 +1,626 @@
+
+    @property
+    def open_report_count(self) -> int:
+        """Returns the count of open reports for this insult."""
+        return self.insultreview_set.filter(status=InsultReview.STATUS.PENDING).count()
+
+    @classmethod
+    def get_by_reference_id(cls, reference_id: str):
+        for prefix in INSULT_REFERENCE_ID_PREFIX_OPTIONS:
+            if reference_id.startswith(prefix):
+                # Extract base36 part and decode to PK
+                base64_part = reference_id.replace(f"{prefix}_", "")
```

Ruff (0.12.2)

Local variable `base64_part` is assigned to but never used

Remove assignment to unused variable `base64_part`

(F841)